basket/news/utils.py

```diff
@@ -95,14 +100,66 @@ def email_is_blocked(email):
     return False
 
 
+def is_authorized(request, email=None):
+    if has_valid_api_key(request):
+        return True
+
+    if email and has_valid_fxa_oauth(request, email):
+        return True
+
+    return False
+
+
 def has_valid_api_key(request):
     # The API key could be the query parameter 'api-key' or the
     # request header 'X-api-key'.
-
     api_key = (request.REQUEST.get('api-key', None) or
                request.REQUEST.get('api_key', None) or
                request.META.get('HTTP_X_API_KEY', None))
-    return APIUser.is_valid(api_key)
+    if api_key:
+        return APIUser.is_valid(api_key)
+
+    return False
+
+
+def has_valid_fxa_oauth(request, email):
+    if not email:
+        return False
+
+    server_urls = fxa.constants.ENVIRONMENT_URLS.get(settings.FXA_OAUTH_SERVER_ENV)
+    if not server_urls:
+        return False
+
+    # Grab the token out of the Authorization header
+    authorization = request.META.get('HTTP_AUTHORIZATION')
+    if not authorization:
+        return False
+
+    authorization = authorization.split(None, 1)
+    if authorization[0].lower() != 'bearer' or len(authorization) != 2:
+        return False
+
+    token = authorization[1]
+    oauth = fxa.oauth.Client(server_url=server_urls['oauth'])
+    profile = fxa.profile.Client(server_url=server_urls['profile'])
```

Comment:
You could consider making these `Client` instances globals to avoid a small amount of initialization overhead on each request, but I don't think it's too big of a deal.

Reply:
Although I guess that could make it harder to test...

Reply:
It would, but I can probably get around that. I'll see what I can do.